microsoft-edge/webview2/concepts/frames.md

```diff
@@ -0,0 +1,496 @@
+   * [CoreWebView2Frame.IsDestroyed Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2frame#isdestroyed)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2_4::FrameCreated event (add](/microsoft-edge/webview2/reference/win32/icorewebview2_4#add_framecreated), [remove)](/microsoft-edge/webview2/reference/win32/icorewebview2_4#remove_framecreated)
+   * [ICoreWebView2FrameCreatedEventArgs interface](/microsoft-edge/webview2/reference/win32/icorewebview2framecreatedeventargs)
+* [ICoreWebView2Frame::Destroyed event (add](/microsoft-edge/webview2/reference/win32/icorewebview2frame#add_destroyed), [remove)](/microsoft-edge/webview2/reference/win32/icorewebview2frame#remove_destroyed)
+   * [ICoreWebView2Frame::IsDestroyed method](/microsoft-edge/webview2/reference/win32/icorewebview2frame#isdestroyed)
+
+---
+
+See also:
+* [iframes](./overview-features-apis.md#iframes) in _Overview of WebView2 features and APIs_.
+
+
+<!-- ====================================================================== -->
+## Navigating within a frame
+
+After a frame is created, the frame navigates to the frame's source URL.  iframes use navigation and navigation events, such as `FrameNavigationStarting` and `NavigationCompleted`.  When the frame navigates to the source URL, the following navigation events are raised:
+
+* `NavigationStarting`
+* `ContentLoading`
+* `HistoryChanged`
+* `DOMContentLoaded`
+* `NavigationCompleted`
+
+
+<!-- ------------------------------ -->
+#### Frequency of navigating within a frame
+
+Navigation can potentially happen within a frame.  As a simple use case, an `iframe` element's `source` attribute is a URL, such as **wikipedia.com**, and the URL is loaded in an iframe.  Usually the navigation happens immediately after the frame is created.  The `ContentLoading`, `DOMContentLoaded`, and `NavigationCompleted` events are then raised.
+
+The frame itself is navigating.  A webpage navigates to a URL.  Similarly, a frame potentially navigates.
+
+After the frame is created, the frame navigates as driven by your host app.  To monitor what's going on in the main page, events such as `NavigationStarted`, `NavigationCompleted`, and `HistoryChanged` enable the host app to navigate back and forth among frames or webpages.  Frames are navigated to a new URL less often than webpages, but the same navigation style is supported.  The user usually cannot navigate inside a frame, although JavaScript enables that; a frame is usually static regarding navigation.
+
+See also:
+* [Standard sequence of events](./navigation-events.md) in _Navigation events for WebView2 apps_.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2.AddScriptToExecuteOnDocumentCreatedAsync Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.addscripttoexecuteondocumentcreatedasync) - This method can be used for an iframe as well as for a webpage.  Your script must check whether it's in an iframe.
```

Comment:
Above this tab-set, I fixed the link (per below tagging) to actually scroll down to the table that lists all events and links to their API Ref - but, in that article, does that table's links go to Frame.FooEvent, or CoreFrame.FooEvent?  Finding: __

Your suggestion is specifically about Frame.FooEvent x7, not CoreFrame.FooEvent x7 totalling 14+ links which would be hard to maintain.  In that article I had started building such set of links but it grew too big, so I went w/ the 7-row table instead but it might lack frame links.
```
See also:
* [Standard sequence of events](./navigation-events.md#standard-sequence-of-events) in _Navigation events for WebView2 apps_.
```

Reply:
Added links:
NavigationStarting - existing
SourceChanged - also added
ContentLoading - added per V.B.
HistoryChanged - added per V.B.
BasicAuthenticationRequested - also added
DOMContentLoaded - added per V.B.
NavigationCompleted - existing

Reply:
Sorry I missed about that we do not have HistoryChanged in frames? Can we have here only the one which we have in frames? Also I'm thinking how easy it would be to fix documentation if we will add new navigation event for frames - and it looks like it is spread in different places=/